File: flutter_app/test/models/appointment_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:flutter_app/models/appointment.dart';

void main() {
  group('Appointment', () {
    test('creates with required fields', () {
      const appt = Appointment(id: '1', title: 'Test', date: 'Today', time: '9:00 AM');
      expect(appt.id, '1');
      expect(appt.title, 'Test');
      expect(appt.date, 'Today');
      expect(appt.time, '9:00 AM');
    });

    test('has correct defaults', () {
      const appt = Appointment(id: '1', title: 'Test', date: 'Today', time: '9 AM');
      expect(appt.location, '');
      expect(appt.notes, '');
      expect(appt.type, 'appointment');
      expect(appt.status, 'todo');
      expect(appt.actionLabel, isNull);
    });

    test('copyWith replaces fields', () {
      const appt = Appointment(id: '1', title: 'A', date: 'Today', time: '9 AM');
      final copy = appt.copyWith(title: 'B', status: 'done');
      expect(copy.title, 'B');
      expect(copy.status, 'done');
      expect(copy.id, '1');
    });

    test('copyWith preserves original when no args', () {
      const appt = Appointment(
        id: '1', title: 'A', date: 'Today', time: '9 AM',
        location: 'Home', notes: 'Note', type: 'med', status: 'done',
        actionLabel: 'View',
      );
      final copy = appt.copyWith();
      expect(copy.id, appt.id);
      expect(copy.title, appt.title);
      expect(copy.location, appt.location);
      expect(copy.actionLabel, appt.actionLabel);
    });

    test('toString returns readable string', () {
      const appt = Appointment(id: '1', title: 'Eye Exam', date: 'Today', time: '2 PM', status: 'todo');
      expect(appt.toString(), 'Appointment(Eye Exam, 2 PM, todo)');
    });
  });
}

File: react-native-app/src/models/__tests__/Appointment.test.js
import { createAppointment, copyAppointment } from '../Appointment';

describe('Appointment', () => {
  test('creates with required fields and defaults', () => {
    const apt = createAppointment({ id: '1', title: 'Test', date: 'Today', time: '9:00 AM' });
    expect(apt.id).toBe('1');
    expect(apt.title).toBe('Test');
    expect(apt.location).toBe('');
    expect(apt.notes).toBe('');
    expect(apt.type).toBe('appointment');
    expect(apt.status).toBe('todo');
    expect(apt.actionLabel).toBeNull();
  });

  test('creates with all fields', () => {
    const apt = createAppointment({
      id: '2', title: 'Eye Doctor', date: 'Today', time: '10:30 AM',
      location: 'Clinic', notes: 'Bring glasses', type: 'medication',
      status: 'done', actionLabel: 'View',
    });
    expect(apt.location).toBe('Clinic');
    expect(apt.type).toBe('medication');
    expect(apt.status).toBe('done');
    expect(apt.actionLabel).toBe('View');
  });

  test('copyAppointment preserves original and applies overrides', () => {
    const original = createAppointment({ id: '1', title: 'Test', date: 'Today', time: '9:00 AM' });
    const copy = copyAppointment(original, { status: 'done', title: 'Updated' });
    expect(copy.status).toBe('done');
    expect(copy.title).toBe('Updated');
    expect(copy.id).toBe('1');
    expect(original.status).toBe('todo');
  });
});
